src/main/java/org/opensearch/flowframework/transport/GetWorkflowStepTransportAction.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.transport;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.model.WorkflowValidator;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+/**
+ * Transport action to retrieve a the workflow step json
+ */
+public class GetWorkflowStepTransportAction extends HandledTransportAction<WorkflowRequest, GetWorkflowStepResponse> {
+
+    private final Logger logger = LogManager.getLogger(GetWorkflowStepTransportAction.class);
+
+    /**
+     * Instantiates a new GetWorkflowStepTransportAction instance
+     * @param transportService the transport service
+     * @param actionFilters action filters
+     */
+    @Inject
+    public GetWorkflowStepTransportAction(TransportService transportService, ActionFilters actionFilters) {
+        super(GetWorkflowStepAction.NAME, transportService, actionFilters, WorkflowRequest::new);
+    }
+
+    @Override
+    protected void doExecute(Task task, WorkflowRequest request, ActionListener<GetWorkflowStepResponse> listener) {
+        try {
+            listener.onResponse(new GetWorkflowStepResponse(WorkflowValidator.parse("mappings/workflow-steps.json")));
```

Comment:
I see we're still retrieving the definitions from a json file, perhaps it might be better to have these definitions defined in a class, something like the Workflow Resources class. But this is just a suggestion for improvement, won't block this PR

Reply:
Yeah, we should improve this in the future. Another issue is that we're reading/parsing this multiple times just to answer the REST query.  JSON text file parsed to xcontent, parsed to String in the response, parsed back to xcontent to be returned to the user.  A single writeable class could just output the xcontent once.